Automattic/happychat-client
# Use new auth library

**Built on top of https://github.com/Automattic/happychat-client/pull/188**

## Summary
Use the new authentication library, see https://github.com/Automattic/happychat-client/pull/188 for details.

## Testing
### Test standalone authentication (regular oauth flow)
0. Run `npm install` in this branch
1. Run `npm start`
2. Visit http://localhost:9000 and make sure authentication and chats work properly

### Test token authentication
0. Run `npm install` in this branch
1. Run `npm run dev:standalone` in this repo
2. Grab your oauth token from a wpcom request made by the client (http://localhost:9000) it is under `Authorization` header after `Bearer`
3. Update `./targets/standalone/index` to use token authentication
```
window.happychatAuth = { 
	type: AUTH_TYPE_WPCOM_OAUTH_BY_TOKEN, 
	options: { 
		token: 'your-token' 
	} 
};
```
4. Visit http://localhost:9000 and make sure authentication and chats work properly
5. Update `./targets/standalone/index` to use older token authentication
```
window.happychatAuth = { 
	accessToken: 'your-oauth-token'
};
```
6. Visit http://localhost:9000 and make sure authentication and chats work properly

### Test calypso authentication
1. Switch calypso to `try/happychat-client` branch (https://github.com/Automattic/wp-calypso/pull/23630)
2. Link this package to wp-calypso, by following these steps
- run `npm link` in this folder
- run `npm link happychat-client` in calypso folder
3. Run `npm dev:npm`
4. Access http://calypso.localhost:3000/me/chat
5. It should load the happychat-client iframe
